src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1063,6 +1063,9 @@ private FeatureType buildFeatureType(
             SimpleFeatureTypeBuilder tb = new SimpleFeatureTypeBuilder();
             tb.setName(info.getName());
             tb.setNamespaceURI(info.getNamespace().getURI());
+            if (ft.getGeometryDescriptor() != null) {
```

Comment:
So there is a [SimpleFeatureTypeBuilder.init( SimpleFeatureType )](http://docs.geotools.org/latest/javadocs/org/geotools/feature/simple/SimpleFeatureTypeBuilder.html#init-org.opengis.feature.simple.SimpleFeatureType-)  intended to be used when duplicating or modifying a feature type, to avoid these kind of problems.

I would be more comfortable using this method so our code is protected against any changes on the GeoTools side (such as needed to copy user data or something).

Reply:
I just tried that out, ... and it doesn't actually work. It causes both the feature and the schema to use the same geometry, but always uses the first geometry instead of the declared default.

Looking at the implementation of SimpleFeatureTypeBuilder.init(SimpleFeatureType), that method doesn't actually do any special handling of geometry, default or otherwise.